workspaces/src/types/mod.rs

```diff
@@ -214,3 +214,82 @@ impl From<AccessKey> for near_primitives::account::AccessKey {
         }
     }
 }
+
+/// Type representing the `block_id` parameter of an RPC query that can be supplied into
+/// viewing blocks or chunks from the network.
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum BlockId {
+    Height(BlockHeight),
+    Hash(CryptoHash),
+}
+
+impl From<BlockId> for near_primitives::types::BlockId {
+    fn from(id: BlockId) -> Self {
+        match id {
+            BlockId::Height(height) => Self::Height(height),
+            BlockId::Hash(hash) => Self::Hash(near_primitives::hash::CryptoHash(hash.0)),
+        }
+    }
+}
+
+/// Finality of a transaction or block in which transaction is included in. For more info
+/// go to the [NEAR finality](https://docs.near.org/docs/concepts/transaction#finality) docs
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum Finality {
+    /// Optimistic finality
+    None,
```

Comment:
Opinionated: Is none a bit misleading here? Why not Optimistic?

Reply:
I think this is just taken directly from nearcore to stay consistent: https://github.com/near/nearcore/blob/9fc9e318000a3a5cd40d34371ad1113fda6ee9ef/core/primitives/src/types.rs#L22-L32.

Probably not worth it to break the pattern, but I agree that the name is strange

Reply:
Yeah, I'm not sure what's best to keep here, but I'm leaning towards doing `Finality::Optimistic`. In RPC https://docs.near.org/docs/api/rpc#using-finality-param, they specify "optimistic" which is definitely more user-friendly and they're breaking the pattern as well. @frol (tagging you since you had a commit on it) would you happen to know why it's called `Finality::None` instead of `Finality::Optimistic` in nearcore?

Reply:
I was only in charge of exposing those variants via RPC interface and so you can see I turned it into "optimistic" for the public interfaces, and I never questioned the reasoning for this naming internally. Ask @bowenwang1996 who implemented it in https://github.com/near/nearcore/pull/2127, but all-in-all, I think the answer will be "it just happened this way for no specific reason", and given it was not part of the public interface back in the days, we never questioned it. Probably, it is a good time to rename those variants for the public interfaces needs.